pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -634,19 +665,61 @@ int kretprobe__tcp_close(struct pt_regs* ctx) {
     return 0;
 }
 
-SEC("kprobe/udp_sendmsg")
-int kprobe__udp_sendmsg(struct pt_regs* ctx) {
+SEC("kprobe/ip6_make_skb")
+int kprobe__ip6_make_skb(struct pt_regs* ctx) {
+     struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
+     size_t size = (size_t)PT_REGS_PARM4(ctx);
+     u64 pid_tgid = bpf_get_current_pid_tgid();
+
+     size = size - sizeof(struct udphdr);
+
+     conn_tuple_t t = {};
+     if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_UDP)) {
+
+         increment_telemetry_count(udp_send_missed);
+         return 0;
+     }
+
+     log_debug("kprobe/ip6_make_skb: pid_tgid: %d, size: %d\n", pid_tgid, size);
+     handle_message(&t, size, 0);
+     increment_telemetry_count(udp_send_processed);
+
+     return 0;
+}
+
+SEC("kprobe/ip_make_skb")
+int kprobe__ip_make_skb(struct pt_regs* ctx) {
     struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
-    size_t size = (size_t)PT_REGS_PARM3(ctx);
+    size_t size = (size_t)PT_REGS_PARM5(ctx);
     u64 pid_tgid = bpf_get_current_pid_tgid();
 
+    size = size - sizeof(struct udphdr);
+
     conn_tuple_t t = {};
     if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_UDP)) {
-        increment_telemetry_count(udp_send_missed);
-        return 0;
+        struct flowi4* fl4 = (struct flowi4*)PT_REGS_PARM2(ctx);
+        bpf_probe_read(&t.saddr_l, sizeof(u32), ((char*)fl4) + offset_saddr_fl4());
+        bpf_probe_read(&t.daddr_l, sizeof(u32), ((char*)fl4) + offset_daddr_fl4());
```

Comment:
Out of curiosity, why is the size `u32` instead of `u64`? Are we truncating `t.saddr_l` and `t.daddr_l` for a specific reason here?

Reply:
IPv4 addresses are only 32 bits.